src/service/action-impl.js

```diff
@@ -566,19 +583,36 @@ export class ActionService {
    * @param {!Array<{tagOrTarget: string, method: string}>} whitelist
    */
   setWhitelist(whitelist) {
-    this.whitelist_ = whitelist;
+    this.whitelist_ = whitelist.filter(
```

Comment:
Curious why add this?

Reply:
A portion of the removed `queryWhitelist_` checked both for invalid entries in shape (` length < 2`) and was always guaranteed to return with the appropriate structure `{ tagOrTarget: string, method: string }`, thought it would healthy to keep that variant in `setWhitelist`. 

`isActionWhitelisted` also relies on the latter half of the above to be true. Another option would be to remove this filter here in favor of a guard in `isActionWhitelist` on the presence of the used properties, but I figured why keep junk at all?

Reply:
That's fine but soft failure is probably bad here. We should let callers know they messed up ASAP.